src/microbe_stage/Microbe.Contact.cs

```diff
@@ -1167,58 +1356,212 @@ private void OnContactBegin(int bodyID, Node body, int bodyShape, int localShape
                     thisMicrobe.PlaySoundEffect("res://assets/sounds/soundeffects/microbe-collision.ogg"));
             }
         }
+        else if (body is IEngulfable engulfable)
+        {
+            if (thisMicrobe.touchedEntities.Add(engulfable))
+            {
+                thisMicrobe.CheckStartEngulfingOnCandidate(engulfable);
+            }
+        }
     }
 
     private void OnContactEnd(int bodyID, Node body, int bodyShape, int localShape)
     {
         _ = bodyID;
         _ = bodyShape;
 
-        if (body is Microbe microbe)
+        if (body is IEngulfable engulfable)
         {
             // GetMicrobeFromShape returns null when it was provided an invalid shape id.
             // This can happen when re-parenting is in progress.
             // https://github.com/Revolutionary-Games/Thrive/issues/2504
             var hitMicrobe = GetMicrobeFromShape(localShape) ?? this;
 
             // TODO: should this also check for pilus before removing the collision?
-            hitMicrobe.touchedMicrobes.Remove(microbe);
+            hitMicrobe.touchedEntities.Remove(engulfable);
         }
     }
 
-    private void OnBodyEnteredEngulfArea(Node body)
+    /*
+    private void OnBodyEnteredPseudopodRange(Node body)
     {
         if (body == this)
             return;
 
-        if (body is Microbe microbe)
+        if (body is IEngulfable engulfable)
         {
-            // TODO: does this need to check for disposed exception?
-            if (microbe.Dead)
-                return;
+            engulfablesInPseudopodRange.Add(engulfable);
+        }
+    }
 
-            if (otherMicrobesInEngulfRange.Add(microbe))
-            {
-                Invoke.Instance.Perform(CheckStartEngulfingOnCandidates);
-            }
+    private void OnBodyExitedPseudopodRange(Node body)
+    {
+        if (body is IEngulfable engulfable)
+        {
+            engulfablesInPseudopodRange.Remove(engulfable);
         }
     }
+    */
 
-    private void OnBodyExitedEngulfArea(Node body)
+    /// <summary>
+    ///   Attempts to engulf the given target into the cytoplasm. Does not check whether the target
+    ///   can be engulfed or not.
+    /// </summary>
+    private void IngestEngulfable(IEngulfable target, float animationSpeed = 2.0f)
     {
-        if (body is Microbe microbe)
+        if (target.EntityNode.GetParent() == this || target.PhagocytosisStep != PhagocytosisPhase.None)
```

Comment:
When is this called with the target already being parented to this?

Reply:
Just a wait a bit, need to refresh up my memory for reasoning on this.

Reply:
I'm done for tonight so I won't be commenting anymore today. Comments are really nice when you have forgotten why you did something (complicated) to remind you.

Reply:
Indeed, I think I'll also push my last commit very soon and resume tomorrow.

Reply:
Turns out this is longer necessary, this is left over check from early commits where the engulfing logic is significantly different from now.

Reply:
So this can be simplified, which is excellent.